src/main/java/org/apache/commons/io/EndianUtils.java

```diff
@@ -182,13 +194,14 @@ public static short readSwappedShort(final InputStream input) throws IOException
     }
 
     /**
-     * Reads an unsigned integer (32-bit) value from a byte array at a given
-     * offset. The value is converted to the opposed endian system while
-     * reading.
+     * Reads a little endian unsigned integer (32-bit) value from a byte array at a given
```

Comment:
Hi @elharo 

This is slightly confusing to me: The method returns a `long` (64-bit) but we are reading an 32-bit integer? Should we clarify this point? Something like "Read a foo as a bar".

Reply:
Since Java's ints are signed you can't fit the unsigned int values from ~2.2 billion  to ~4 billion into an int, so you have to return a long. The long will always be <= 2^32, but might be greater than Integer.MAX_INT. Not sure how much of that to say here.

Reply:
Makes sense, the comment already mentions "unsigned". TY for the clarification.